app/vmalert/web.go

```diff
@@ -102,23 +114,45 @@ func (rh *requestHandler) handler(w http.ResponseWriter, r *http.Request) bool {
 		w.Write(data)
 		return true
 	case "/vmalert/api/v1/alerts", "/api/v1/alerts":
-		// path used by Grafana for ng alerting
-		data, err := rh.listAlerts()
+		if r.FormValue(paramGroupID) == "" && r.FormValue(paramAlertID) == "" {
+			// show all alerts
+			// path used by Grafana for ng alerting
+			data, err := rh.listAlerts()
+			if err != nil {
+				httpserver.Errorf(w, r, "%s", err)
+				return true
+			}
+			w.Header().Set("Content-Type", "application/json")
+			w.Write(data)
+			return true
+		}
+
+		// show status for specific alert
+		alert, err := rh.getAlert(r)
 		if err != nil {
 			httpserver.Errorf(w, r, "%s", err)
 			return true
 		}
+		data, err := json.Marshal(alert)
+		if err != nil {
+			httpserver.Errorf(w, r, "failed to marshal alert: %s", err)
+			return true
+		}
 		w.Header().Set("Content-Type", "application/json")
 		w.Write(data)
 		return true
-
 	case "/-/reload":
 		logger.Infof("api config reload was called, sending sighup")
 		procutil.SelfSIGHUP()
 		w.WriteHeader(http.StatusOK)
 		return true
 
 	default:
+		// Support of deprecated links:
```

Comment:
What is the point for leaving the support for old links here? If these links are generated only by `vmalert` itself, then it is safe removing the support for old links right now.

Reply:
Just in case someone built some additional automation for this. It costs nothing by now for support old links to us.

Reply:
This increases technical debt for maintaining deprecated code paths.

Reply:
Sure! I've tried to isolate deprecated logic, so removing it in future should be easy.